specifyweb/frontend/js_src/lib/components/DataModel/types.ts

```diff
@@ -4956,6 +4956,7 @@ export type SpAppResourceDir = {
     readonly timestampModified: string | null;
     readonly userType: string | null;
     readonly version: number | null;
+    readonly scope: string;
```

Comment:
I don't see the `scope` field in the response from http://localhost/api/specify/spappresourcedir/?limit=0. can you double check why this was added?

Reply:
oh I see what you are doing.
the `scope` field is added on the front-end only.
this file defines types for data we receive from and send to back-end only!

what is more, this file is auto-generated - manual changes should be minimized. see:
https://github.com/specify/specify7/blob/issue-4170/specifyweb/frontend/js_src/lib/components/DataModel/types.ts#L1-L24

perhaps you meant to use the ScopedAppResourceDir type instead of modifying this type?

Reply:
sorry for the misunderstanding. let me clarify this further:

'scope' field is not coming from the back-end (database/python)
it is added by the front-end
[types.ts](https://github.com/specify/specify7/blob/issue-4170/specifyweb/frontend/js_src/lib/components/DataModel/types.ts#L1-L24) file is only for fields coming from the back-end
'scope' is added by the front-end
so 'scope' field does not belong in the types.ts file.

specifically, see this part - https://github.com/specify/specify7/blob/6e0b18b3b4612787d1de74f01780bb56dc92fd11/specifyweb/frontend/js_src/lib/components/AppResources/hooks.ts#L41-L45

we fetch the SpAppResourceDir from the back-end
and then we add the scope field.

the types.ts file is responsible for telling TypeScript what to expect to receive from the back-end API (i.e what `fetchCollection`/`fetchResource`/`new table.CollectionObject.Resource({id: someId}).fetch()` would return)
because of the code change you did, typescript now things that `fetchCollection` would include the scope field - IT DOES NOT. The scope field is added later by the front-end.

Does this part make sense?

That is why you should not be modifying the [types.ts](https://github.com/specify/specify7/blob/issue-4170/specifyweb/frontend/js_src/lib/components/DataModel/types.ts#L1-L24) file. It is auto-generated from the database schema. see the comment at the top of that file for more information:

https://github.com/specify/specify7/blob/issue-4170/specifyweb/frontend/js_src/lib/components/DataModel/types.ts#L1-L24

Now, if you check the place on the front-end where the scope field is added (the [useAppResources](https://github.com/specify/specify7/blob/issue-4177/specifyweb/frontend/js_src/lib/components/AppResources/hooks.ts#L34-L66) hook), you will see that the return type of the hook uses this type for directories:

https://github.com/specify/specify7/blob/issue-4177/specifyweb/frontend/js_src/lib/components/AppResources/hooks.ts#L26

See how it uses ScopedAppResourceDir rather than AppResourceDir?

Which is defined as:

https://github.com/specify/specify7/blob/issue-4177/specifyweb/frontend/js_src/lib/components/AppResources/types.tsx#L39C13-L41

SO what you had to do in this case is to change the argument type of the EditAppResource component to expect the ScopedAppResourceDir type rather than AppResourceDir if you need to use the `scope` field (and resolve the upstream typescript errors resulting from that if in places where EditAppResource is called, it receives  AppResourceDir rather than ScopedAppResourceDir)

Reply:
Thank you for details :+1: